From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Fri, 24 Jul 2026 19:24:02 +0200
Subject: dt-bindings: usb: allwinner,sun4i-a10-musb: Allow idma region on A83T
 and H3

The A83T and H3/H5 have the same USB-OTG internal DMA engine as the A64,
in the same location above the core registers. Allow describing its
register region on those SoCs too.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 Documentation/devicetree/bindings/usb/allwinner,sun4i-a10-musb.yaml | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Documentation/devicetree/bindings/usb/allwinner,sun4i-a10-musb.yaml b/Documentation/devicetree/bindings/usb/allwinner,sun4i-a10-musb.yaml
index 111111111111..222222222222 100644
--- a/Documentation/devicetree/bindings/usb/allwinner,sun4i-a10-musb.yaml
+++ b/Documentation/devicetree/bindings/usb/allwinner,sun4i-a10-musb.yaml
@@ -99,14 +99,17 @@ allOf:
       required:
         - resets
 
-  # Only the A64 has the vendor "internal DMA" engine, and hence a second
-  # register region.
+  # Only the A83T, H3 (and its H5 twin) and A64 have the vendor "internal
+  # DMA" engine, and hence a second register region.
   - if:
       not:
         properties:
           compatible:
             contains:
-              const: allwinner,sun50i-a64-musb
+              enum:
+                - allwinner,sun8i-a83t-musb
+                - allwinner,sun8i-h3-musb
+                - allwinner,sun50i-a64-musb
     then:
       properties:
         reg:
-- 
Armbian

